### ISSUE TITLE
Require full URL for absolute URLs in Link fields

### ISSUE THREAD

**Issue by quicksketch** [backdrop/backdrop-issues#1159]
Followup to the addition of Link module in #240. From that issue:

The only thing I'd like to see changed at this point is the "optional" treatment of the protocol. Traditionally, you've been able to omit the "http://" from absolute URLs. e.g. "www.google.com" would be considered an absolute URL and be displayed as http://www.google.com. The trouble with this is that now that there are so many TLDs, it's possible that we'll end up with an overlap between file extensions and TLDs. For example, there could be a TLD for "rss" or "php" (the .js domain is actually a [real attempt](https://github.com/ozten/TLD.js/issues/8)). A relative URL such as "index.php" will currently be interpreted as an absolute URL, i.e. http://index.php/ We should remove the ability for the protocol to be optional IMO, as the current guessing is now unreliable.

**Comment by BWPanda** [comment]
@quicksketch I tried playing with this, and I found and commented out the code where the `http://` is added to external links without one already. However that results, when I enter a URL of `panda.id.au`, in this: `http://sandbox.lndo.site/panda.id.au`

How should this situation be handled then? I think it's too late to raise a validation error on the form...

Here's a POC PR to test: https://github.com/backdrop/backdrop/pull/3255

**Comment by klonos** [comment]
If we end up leaving the protocol adding in place, should we change the default from `http` to `https`?

**Comment by BWPanda** [comment]
What if we check the URL? Like get the response when accessing it... If it works without redirection, then great (in which case, yes, prefer `https`). If not, then...?

**Comment by klonos** [comment]
> Like get the response when accessing it

Yeah, that could work, but not in cases where the person is testing offline ...but then again, what are the chances of that? 🤔 

My concern is that most (all?) modern browsers throw errors/warnings when there's "mixed" http and https content. My assumption is that most sites have adapted, in order to avoid being flagged as unsecure.

My understanding is that we moved from an era where having a secure site was the exception:

![image](https://user-images.githubusercontent.com/2423362/90561269-2124bd80-e1a9-11ea-885b-96e007586910.png)

...to the opposite, where https is the expected default, and anything else is flagged as insecure.

If this is just a Chrome thing, or not true entirely, then feel free to ignore me 😅
